soroban-env-host/src/test/lifecycle.rs

```diff
@@ -1140,5 +1142,193 @@ mod cap_54_55_56 {
         Ok(())
     }
 
+    // Test that, when running on protocol vNew and calling a wasm that is not
+    // in the footprint or storage, we just get a storage error, not any kind of
+    // internal error. This is a legitimate (if erroneous) way to invoke the host
+    // from outside and we should fail gracefully.
+    #[test]
+    fn test_v_new_call_nonexistent_wasm() -> Result<(), HostError> {
+        let (host, contract_id) = upload_and_make_host_for_next_ledger(
+            "test_v_new_call_nonexistent_wasm_upload",
+            V_NEW,
+            "test_v_new_call_nonexistent_wasm_call",
+            V_NEW,
+        )?;
+        let contract = host.add_host_object(crate::xdr::ScAddress::Contract(contract_id))?;
+
+        // Remove the wasm from the storage and footprint.
+        let wasm_to_delete = wasm_module_with_a_bit_of_everything(V_NEW);
+        let wasm_hash = Hash(
+            sha256_hash_from_bytes(&wasm_to_delete, host.as_budget())?
+                .try_into()
+                .unwrap(),
+        );
+        let wasm_key = host.contract_code_ledger_key(&wasm_hash)?;
+        host.with_mut_storage(|storage| {
+            let budget = host.budget_cloned();
+            storage.footprint.0 = storage
+                .footprint
+                .0
+                .remove::<Rc<LedgerKey>>(&wasm_key, &budget)?
+                .unwrap()
+                .0;
+            storage.map = storage
```

Comment:
Can we also have a test case where the entry is `None` in map, i.e. Wasm is in the footprint, but doesn't exist? The expected error should be `Storage, MissingValue` (I believe the code is correct, but it's nice to have a supporting test as well).

Reply:
sure I'll tack one on